sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -96,6 +98,11 @@ private[sql] class SessionState(
     hadoopConf
   }
 
+  def newHadoopConfWithCaseInsensitiveOptions(options: CaseInsensitiveStringMap): Configuration = {
```

Comment:
I don't think we should pollute `SessionState` with the case insensitive map stuff. Can we inline this method?

Reply:
Otherwise, developers might not be aware of using `.getOriginalMap` if they want to create Hadoop configuration from CaseInsensitiveStringMap.

Reply:
Then we should document it in `CaseInsensitiveMap`. data source developers can't access `SessionState`

Reply:
I agree with @cloud-fan, I'd rather use `newHadoopConfWithOptions` and not add a new method here.